src/ralph/ralph2_sync/subscribers.py

```diff
@@ -105,6 +106,58 @@ def _get_configuration_path_from_venture_role(venture_role_id):
     return None
 
 
+def _handle_management_ip(dca, management_ip, management_hostname):
+    if management_ip:
+        try:
+            ip = IPAddress.objects.get(address=management_ip)
+        except IPAddress.DoesNotExist:
+            dca.management_ip = management_ip
+        else:
+            ip.ethernet.base_object = dca
+            ip.ethernet.save()
+            ip.is_management = True
+            ip.save()
+        dca.management_hostname = management_hostname
+        dca.save()
+    else:
+        del dca.management_ip
+
+
+def _handle_ips(obj, ips):
+    if ips is None:
+        return
+    management_ip = list(filter(lambda ip: ip['is_management'], ips))
+    if len(management_ip) > 1:
+        logger.error('More than one management IP ({}) for {}.'.format(
+            ', '.join([ip['address'] for ip in management_ip]), obj
+        ))
+    elif len(management_ip) == 1:
+        management_ip = management_ip[0]
+        _handle_management_ip(
+            obj, management_ip['address'], management_ip['hostname']
+        )
+    for ip_dict in filter(lambda ip: not ip['is_management'], ips):
+        ip, created = IPAddress.objects.get_or_create(
+            address=ip_dict['address'],
+            defaults=dict(
+                hostname=ip_dict['hostname'],
+                dhcp_expose=ip_dict['dhcp_expose'],
+                is_management=False
+            )
+        )
+        if created:
+            ip.ethernet, _ = Ethernet.objects.get_or_create(
+                base_object=obj, mac=ip_dict['mac']
```

Comment:
if `mac` is empty and there is more than one IP without mac assigned to this obj, there will be `MultipleObjectsReturned` raised.